### DISCUSSION TITLE
SKPicture changed after SvgDocument DeepCopy

### DISCUSSION THREAD

**Discussion by mzh3511** [wieslawsoltes/Svg.Skia#160] [Q&A] [answered]
Hello, I found an issue during working with the `Svg.Skia`, it seems the `SvgElement.DeepCopy` missing `_styles`.

I have an SVG file attached ![Demo](https://user-images.githubusercontent.com/3412782/237051878-00c7f2aa-60d9-47db-9d41-f654a0ebcd88.svg)

Get the `SKPicture` from the `SvgDocument` of the above SVG file, then render it by `SkiaSharp`, the screenshot is 
![Not deep copy screenshot](https://user-images.githubusercontent.com/3412782/237052759-d4201561-870b-4f55-916d-c06395e0db67.jpg)

The following are my codes
```
        private void Form1_Load(object sender, EventArgs e)
        {
            var svgDocument = SvgExtensions.Open(@"G:\Svg Files\Demo.svg");
            if (svgDocument != null)
            {
                // svgDocument = (SvgDocument)svgDocument.DeepCopy<SvgDocument>();

                DoColorReplace(svgDocument);

                var settings = new SKSvgSettings();
                var skiaModel = new SkiaModel(settings);
                var assetLoader = new SkiaAssetLoader(skiaModel);

                var shimPicture = SvgExtensions.ToModel(svgDocument, assetLoader, out var drawable, out _);
                _skiaPicture = skiaModel.ToSKPicture(shimPicture);
            }
        }
```

Then I do not use the `SvgDocument` of the above SVG file directly, I use its `DeepCopy` result, I get the wrong screenshot
![Deep copy screenshot](https://user-images.githubusercontent.com/3412782/237053183-9ee59263-62a8-479a-bc82-c1c78c15e0bf.png)

I checked the codes and found the issue is because of missing `_style` in the `SvgElement`, the change screenshot:
![Fix issue](https://user-images.githubusercontent.com/3412782/237055160-1c29c942-0142-4f6d-964d-15856f0f63ee.png)

Can you double-check it, fix the issue and publish a new NuGet package, please? 
Thanks very much

**Comment by mzh3511** [comment]
Hello, I know that `Svg.Custom` is from your `SVG` code repository, which is forked from [svg-net](https://github.com/svg-net/SVG).

Now, the `svg-net` fixed the `DeepCopy` issue ([PR1053 ](https://github.com/svg-net/SVG/pull/1053)), can you merge it in and publish a new NuGet package, please?

**Reply by wieslawsoltes** [reply]
Will do release today or tomorrow

**Reply by wieslawsoltes** [reply, marked as answer]
Released
